src/modules/claim/services/claim-issuance.service.ts

```diff
@@ -83,16 +86,70 @@ export class ClaimIssuanceService {
       claimType: dto.claimType,
     });
 
-    await this.roleService.verifyEnrolmentPrecondition({
-      claimType: dto.claimType,
-      userDID: dto.requester,
-    });
+    const enrolmentPreconditions =
+      await this.roleService.fetchEnrolmentPreconditions({
+        claimType: dto.claimType,
+      });
+    if (
+      enrolmentPreconditions?.length > 0 &&
+      enrolmentPreconditions.find((precond) => precond.type === 'role')
+    )
+      await this.verifyEnrolmentPrerequisites(
+        enrolmentPreconditions,
+        dto.requester,
+        claimType
+      );
 
     await this.createAndIssue(dto, dto.requester);
 
     return ClaimHandleResult.Success();
   }
 
+  public async verifyEnrolmentPrerequisites(
+    enrolmentPreconditions: {
+      type: PreconditionType;
+      conditions: string[];
+    }[],
+    requester: string,
+    claimType: string
+  ) {
+    for (const { type, conditions } of enrolmentPreconditions) {
+      if (type === 'role' && conditions?.length > 0) {
```

Comment:
I think this could be seen as a duplication of the check above (the `enrolmentPreconditions.find`). I think probably only one of them are needed